update pattern-not to include references to additional options (#2210)
semgrep/semgrep-docs: docs/writing-rules/rule-syntax.md

db_query("SELECT * FROM ...", verify=True, env="prod")
```

> Additionally, `pattern-not` accepts one `pattern` or `patterns` property, which negates the entire `pattern` or `patterns` construction.

Reviewer: I don't love this example but I think it's clear enough in context!

I'm also not sure how to describe the unit of allowable constructions here that includes `pattern` and `patterns` and `pattern-either` (and maybe even `pattern-not` although that would be silly). Opted to suggest removing `pattern` partly for brevity but also because there's no point in using just one pattern this way (you could just use the basic `pattern-not` syntax instead).

Swapped "additionally" for "alternatively" because I think it's one or the other, not both.

Nota bene that the fix suggestion doesn't seem to be handling the closing backticks for the interior code block correctly (might need fixing if you take this suggestion).

```suggestion
Alternatively, `pattern-not` accepts a `patterns` or `pattern-either` property, and negates everything inside it.

```yaml
rules:
  - id: unverified-db-query
    patterns:
      - pattern: db_query(...)
      - pattern-not:
          pattern-either:
            - pattern: db_query(..., verify=True, ...)
            - pattern-inside: |
                with ensure_verified(db_query):
                  db_query(...)
    message: Found unverified db query
    severity: ERROR
    languages:
      - python
```
```
